### PR TITLE
Add note about Result to `find_or_last` and `find_or_first`

### PR DESCRIPTION
**Pull request by axelkar**
See GH-983

@phimuemue 

Is the wording okay? What about the explicit type definition?

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -2000,6 +2000,21 @@ pub trait Itertools: Iterator {
     /// assert_eq!(numbers.iter().find_or_last(|&&x| x > 2), Some(&3));
     /// assert_eq!(std::iter::empty::<i32>().find_or_last(|&x| x > 5), None);
     /// ```
+    ///
+    /// An iterator of [`Result`]s can find the first `Ok` or the last `Err`:
```
Comment (phimuemue) [inline src/lib.rs:2004]: I think we can make it less prominent. Maybe just add the following after the previous `assert_eq`s:

```
// find first Ok if present, or return the last Err:
assert_eq!(vec![Err(1), Ok(2), Err(3), Ok(4)].find_or_last(Result::is_ok), Ok(2));
assert_eq!(vec![Err(1), Err(2)].find_or_last(Result::is_ok), Err(2));
...
```

**Comment by phimuemue** [comment]
> What about the explicit type definition?

We can use them where needed, otherwise elide them.

**Comment by phimuemue** [comment]
@axelkar There are some CI errors. Ours should have been fixed by https://github.com/rust-itertools/itertools/commit/4f49f2636aa95283cc4c921e6a3e04efe317fda5. Could you rebase and fix yours if needed?

File: src/lib.rs [hunk 2]
```diff
@@ -1999,6 +1999,15 @@ pub trait Itertools: Iterator {
     /// assert_eq!(numbers.iter().find_or_last(|&&x| x > 5), Some(&4));
     /// assert_eq!(numbers.iter().find_or_last(|&&x| x > 2), Some(&3));
     /// assert_eq!(std::iter::empty::<i32>().find_or_last(|&x| x > 5), None);
+    ///
+    /// // An iterator of Results can return the first Ok or the last Err:
+    /// let input = vec![Err(()), Ok(11), Err(()), Ok(22)];
+    /// assert_eq!(.into_iter().find_or_last(Result::is_ok), Some(Ok(11)));
```
Comment (phimuemue) [inline src/lib.rs:2005]: Should this be `assert_eq!(input.into_iter().find_or_last(Result::is_ok), Some(Ok(11)));`?

**Comment by phimuemue** [comment]
Thanks a bunch.

File: src/lib.rs [hunk 3]
```diff
@@ -2027,6 +2036,15 @@ pub trait Itertools: Iterator {
     /// assert_eq!(numbers.iter().find_or_first(|&&x| x > 5), Some(&1));
     /// assert_eq!(numbers.iter().find_or_first(|&&x| x > 2), Some(&3));
     /// assert_eq!(std::iter::empty::<i32>().find_or_first(|&x| x > 5), None);
+    ///
+    /// // An iterator of Results can return the first Ok or the last Err:
```
Comment (phimuemue) [inline src/lib.rs:2040]: This should be "or the first Err". Sorry for the back-and-forth, I did not realize it the first time.

**Review by phimuemue — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 3 above)
Comment (axelkar) [reply src/lib.rs]: It's really something I overlooked. Thanks for reviewing!

**Comment by phimuemue** [comment]
Thanks.

**Review by phimuemue — APPROVED** [review]

**Pull request merged**